### PR TITLE
Add UTXO changes full API

### PR DESCRIPTION
**Pull request by thibault-martinez**
https://github.com/iotaledger/iota.go/pull/638

### CODE REVIEW

File: bindings/core/src/method/client.rs [hunk 1]
```diff
@@ -273,6 +273,12 @@ pub enum ClientMethod {
...
         commitment_id: SlotCommitmentId,
     },
+    /// Get all full UTXO changes of a given slot by Commitment ID.
+    #[serde(rename_all = "camelCase")]
+    GetUtxoChangesFull {
+        /// Commitment ID of the commitment to look up.
+        commitment_id: SlotCommitmentId,
+    },
     /// Look up a commitment by a given commitment index.
     GetCommitmentByIndex {
```
Comment (kwek20) [inline bindings/core/src/method/client.rs:283]: order! move this one up to behind here please:
```
GetCommitment {
        /// Commitment ID of the commitment to look up.
        commitment_id: SlotCommitmentId,
    },
```

Comment (thibault-martinez) [reply bindings/core/src/method/client.rs:283]: I'd rather not, this is the order of the TIP

Comment (kwek20) [reply bindings/core/src/method/client.rs:283]: why wouldnt we have it alphabetical? what order does the tip even follow?

Comment (thibault-martinez) [reply bindings/core/src/method/client.rs:283]: They group by parameter, first SlotCommitmentId then SlotIndex

**Review by Thoralf-M — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

File: sdk/src/client/node_api/core/routes.rs [hunk 2]
```diff
@@ -333,6 +334,17 @@ impl ClientInner {
         self.get_request(path, None, false, false).await
     }
 
+    /// Get all full UTXO changes of a given slot by slot commitment ID.
+    /// GET /api/core/v3/commitments/{commitmentId}/utxo-changes/full
+    pub async fn get_utxo_changes_full_by_slot_commitment_id(
```
Comment (Alex6323) [inline sdk/src/client/node_api/core/routes.rs:339]: I would suggest the name `get_full_utxo_changes_by_slot_commitment_id` tbh, but ignore if you disagree.
Should you agree, then the response type should probably be renamed too.

File: bindings/core/src/response.rs [hunk 3]
```diff
@@ -142,6 +143,10 @@ pub enum Response {
     /// - [`GetUtxoChangesByIndex`](crate::method::ClientMethod::GetUtxoChangesByIndex)
     UtxoChanges(UtxoChangesResponse),
     /// Response for:
+    /// - [`GetUtxoChangesFull`](crate::method::ClientMethod::GetUtxoChangesFull)
+    /// - [`GetUtxoChangesFullByIndex`](crate::method::ClientMethod::GetUtxoChangesFullByIndex)
+    UtxoChangesFull(UtxoChangesFullResponse),
```
Comment (Alex6323) [inline bindings/core/src/response.rs:148]: btw ... I think for the variant it's reasonable to have `full` suffixed as it is here so both variants can be next to each other, although maybe we should just no alphabetially sort enum variants, but group them logically :shrug:

File: sdk/src/client/node_api/core/routes.rs (hunk 2 above)
Comment (thibault-martinez) [reply sdk/src/client/node_api/core/routes.rs:339]: I would keep this way so it's more consistent with the URL

File: bindings/core/src/response.rs (hunk 3 above)
Comment (thibault-martinez) [reply bindings/core/src/response.rs:148]: They are grouped logically, by parameter. See Brord's comment.

Comment (thibault-martinez) [reply bindings/core/src/response.rs:148]: Oh that's the response file actually, that's a bit different. Don't think it matters that much...

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**